nixos/modules/services/networking/juicity.nix

```
@@ -0,0 +1,108 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+let
+  inherit (lib)
+    mkOption
+    types
+    mkPackageOption
+    nameValuePair
+    mkEnableOption
+    mapAttrs'
+    optional
+    mkIf
+    ;
+  cfg = config.services.juicity;
+in
+{
+  meta = {
+    maintainers = with lib.maintainers; [ oluceps ];
+  };
+
+  options.services.juicity = {
+    instances = mkOption {
+      description = "list of juicity instance";
+      type = types.attrsOf (
+        types.submodule {
+          options = {
+            enable = mkEnableOption "enable this juicity instance";
+            package = mkPackageOption pkgs "juicity" { };
+            credentials = mkOption {
+              type = types.listOf types.str;
+              default = [ ];
+              description = ''
+                Load extra credentials.
+                Could be written as systemd `LoadCredentials` format e.g.
+                `["key:/etc/juicity-key"]` and access in config with
+                `/run/credentials/juicity-$\{name}.service/key`
+              '';
+            };
+            openFirewall = mkOption {
+              type = with types; nullOr port;
+              default = null;
+              description = "Open firewall port";
+            };
+            serve = mkEnableOption "using `juicity-server` instead of `juicity-client`";
+            configFile = mkOption {
+              type = types.str;
+              default = "/etc/juicity/server.json";
+              description = "Config file location, absolute path";
+            };
+          };
+        }
+      );
+      default = { };
+    };
+  };
+
+  config = mkIf (cfg.instances != { }) {
+    environment.systemPackages = lib.unique (
+      lib.foldr (s: acc: acc ++ (optional s.enable s.package)) [ ] (builtins.attrValues cfg.instances)
+    );
```

I believe `environment` handles deduplication by itself?

```suggestion
    environment.systemPackages = 
      lib.concatMap (s: optional s.enable s.package) (builtins.attrValues cfg.instances);
```